server/app/pagination.py

```diff
@@ -0,0 +1,240 @@
+        return {
+            'limit': self.items_per_page,
+            'offset': self.items_per_page * (page - 1)
+        }
+
+    def _template_parameters(self, item_count: int,
+            url: Optional[str]=None, page:Optional[int]=None) -> PaginationDict:
+        '''Return parameters for paginate template.'''
+        url = Pagination._current_url() if url is None else url
+        if page is None:
+            current = Pagination._current_page(url)
+        else:
+            current = page
+        return {
+            'url': url,
+            'page_current': current,
+            'page_count': self._page_count(item_count),
+            'items_count': item_count,
+            'items_current': self._page_item_count(item_count, current)
+
+        }
+
+    def template_kwargs(self, item_count: int,
+            url: Optional[str]=None, page:Optional[int]=None
+        )-> TemplatePaginationDict:
+        '''Return parameters for pagination template bounded with dict with
+            expected key name by template'''
+        return {
+            'pagination': self._template_parameters(
+                item_count, url, page
+            )
+        }
+
+def use_page(f):
+    '''
+    Decorator for retrive page number from request. Part of medium-level
+    pagination solution.
+
+    Checklist (what you need to do to use it correctly):
+    1. Add this decorator to your action (without parenthness)
+    2. Take as function argument "page"
+    3. Create object of Pagination class
+    4. Calculate limit and offset
+    5. Pass limit and offset to specific repository read method by unpack
+        content by duble asteriks
+    6. Fetch total count of your items
+    7. Create template pagination arguments
+    8. Pass template arguments by unpack it using double asteriks
+        to "render_template" function
+    9. Include "paginate.html" template in your template
+
+    Example
+    =======
+    >>> # In action .py file. In comments are checklist steps.
+    >>> # 1. Use decorator
+        @use_page
+        # 2. Take parameter
+        def obslist(page: int):
+            # 3. Create pagination object
+            pagination = Pagination()
+            ...
+            # 4. Calculate limit and offset
+            limit_and_offset = pagination.limit_and_offset(page)
+            # 5. Pass limit and offset to read item method
+            items = repo.read_items(**limit_and_offset)
+            # 6. Fetch item count
+            count = repo.count_items()
+            # 7. Create template arguments dict
+            p_args = pagination.template_kwargs(observation_count)
+            # 8. Pass template arguments
+            return render_template('template.html', items=items, **p_args)
+
+    >>> # 9. Include pagination template in your .html template file
+    >>> {% include 'pagination.html' %}
+    '''
+    return use_kwargs({
+        PARAM_NAME: webargs.fields.Int(missing=1, 
```

Comment:
From PEP8, official Python style guide:
>Limit all lines to a maximum of 79 characters.
>
>For flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters.

I think that normal lines should have max 80 character, if you have good reason then it may be longer, but avoid line longer then 120 characters.
I often have two windows side-by-side and this limitations are useful for me.

Reply:
I dislike overly wrapped text. It causes less code to fit on the screen and you need to scroll more. Anyway, this is just a personal preference, so let's keep the code as it is.